server/utils/internal/api_handler.py

```
@@ -66,32 +63,125 @@ async def _handle_kubectl_execute(headers, body, writer):
+            await _send_json_response(writer, result)
             return
-        
-        await websocket.send(json.dumps({'type': 'kubectl_command', 'command_id': command_id, 'command': command, 'timeout': timeout}))
-        result = await asyncio.wait_for(future, timeout=timeout)
-        await _send_json_response(writer, result)
-    except asyncio.TimeoutError:
-        await _send_json_response(writer, {'success': False, 'error': 'No response from agent'})
     except Exception as e:
-        logger.error(f"Error communicating with kubectl agent: {e}", exc_info=True)
-        await _send_json_response(writer, {'success': False, 'error': f'Agent communication error: {str(e)}'})
+        logger.error("Kubeconfig fallback failed for cluster %s: %s", cluster_id, e, exc_info=True)
```

Ruff (0.15.15)

[warning] Logging `.exception(...)` should be used instead of `.error(..., exc_info=True)`

(G201)